### PR TITLE
Enhancement to add the document language to DOCX based on the report locale (#1620)

### PR DESCRIPTION
**Pull request by speckyspooky**
Enhancement to add the document language to DOCX based on the report locale (#1620)

### CODE REVIEW

File: engine/org.eclipse.birt.report.engine.emitter.docx/src/org/eclipse/birt/report/engine/emitter/docx/writer/Document.java
Comment (hvbtup) [inline engine/org.eclipse.birt.report.engine.emitter.docx/src/org/eclipse/birt/report/engine/emitter/docx/writer/Document.java:1]: I don't understand the two dozens of deleted lines here. They seem unrelated to the language change.

Comment (hvbtup) [inline engine/org.eclipse.birt.report.engine.emitter.docx/src/org/eclipse/birt/report/engine/emitter/docx/writer/Document.java:1]: And I'd prefer (only if MS Word accepts it) if it was possible to *not* set a language, e.g. skip the whole `w:lang` element, instead of using the old `en-US` default (which is a minor bug IMO).

File: engine/org.eclipse.birt.report.engine.emitter.wpml/src/org/eclipse/birt/report/engine/emitter/wpml/AbstractEmitterImpl.java
Comment (hvbtup) [inline engine/org.eclipse.birt.report.engine.emitter.wpml/src/org/eclipse/birt/report/engine/emitter/wpml/AbstractEmitterImpl.java:1]: I assume that the `locale` property inside the rptdesign is already used to initialize `reportContext.getLocale()` ?

**Review by hvbtup — COMMENTED** [review]
The PR seems to contain an change in one file (see my comments). Can you explain or remove this deletion block?

Personally I think the default en-US should be changed to if possible, but that's a matter of taste, so since you wrote the code, you may decide yourself.

And one question regarding using `reportContext.getLocale()` as the source. It seems to work, but at first glance I couldn't find where the rptdesign's locale property is used to initialize the reportContext's locale.

File: engine/org.eclipse.birt.report.engine.emitter.docx/src/org/eclipse/birt/report/engine/emitter/docx/writer/Document.java
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine.emitter.docx/src/org/eclipse/birt/report/engine/emitter/docx/writer/Document.java:1]: The deleted code is clean-up of old commented code.

Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine.emitter.docx/src/org/eclipse/birt/report/engine/emitter/docx/writer/Document.java:1]: I will not change the structure over all because we can get effects if we remove the "w:lang" over all.
In my understanding this tag is a default tag of the styles of Word. Therefore I will not remove them.

Comment (hvbtup) [reply engine/org.eclipse.birt.report.engine.emitter.docx/src/org/eclipse/birt/report/engine/emitter/docx/writer/Document.java:1]: OK.

File: engine/org.eclipse.birt.report.engine.emitter.wpml/src/org/eclipse/birt/report/engine/emitter/wpml/AbstractEmitterImpl.java
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine.emitter.wpml/src/org/eclipse/birt/report/engine/emitter/wpml/AbstractEmitterImpl.java:1]: The "locale"-property is a new one and will be initialized in the "public void initialize()"-method.
This was a local variable before and I use this one for the language. So I avoid special handling if the local is null and use the same behavior.

Comment (hvbtup) [reply engine/org.eclipse.birt.report.engine.emitter.wpml/src/org/eclipse/birt/report/engine/emitter/wpml/AbstractEmitterImpl.java:1]: Yes, I understood that already. My question is from the perspective that we said we want to use the locale property *from the rptdesign file* as the source to set the language inside the DOCX file. 
And in the PR I see that we use the locale property from `reportContext` as the source instead.
So there's kind of a missing link from "rptdesign.report.locale" to "reportContext.locale", which I assume is existing somewhere in the existing source (outside of the PR). but I couldn't find it. I only want to be sure.

Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine.emitter.wpml/src/org/eclipse/birt/report/engine/emitter/wpml/AbstractEmitterImpl.java:1]: Ohh, well, this is a good question. On of my test cases was to enter a local on report side "test-locale" & "auto"
and I got "test-locale" and "auto" internally which will be switched to "en" because these are no valid locale-values.
I tested also the option of "blank" locale and the result was the fallback "en" again.

My test environment is MS Windows 11 "de". So the system-locale wasn't used on my side.
But I cannot say directly where the transfer will be done on code level.

Comment (hvbtup) [reply engine/org.eclipse.birt.report.engine.emitter.wpml/src/org/eclipse/birt/report/engine/emitter/wpml/AbstractEmitterImpl.java:1]: OK.

**Comment by speckyspooky** [comment]
Is your "OK" the approval for that change then I would merge the change.

**Comment by hvbtup** [comment]
Yes, pls merge.
